vmm_core/virt_support_apic/src/lib.rs

```diff
@@ -1068,6 +1068,17 @@ impl<T: ApicClient> LocalApicAccess<'_, T> {
         }
     }
 
+    /// Pull offloaded APIC state and rewind the given vector.
+    pub fn rewind_offloaded_interrupt(&mut self, vector: u8) {
+        self.ensure_state_local();
+        let local_isr_top = self.apic.isr.pop();
+        assert_eq!(local_isr_top, Some(vector));
```

Comment:
This is not true with auto EOI. We need to extend the saved state to remember whether the last interrupt was auto EOI, or something. See hypervisor/legacy HCL for how this is handled.

Reply:
We're only rewinding when we're offloaded though, so we won't have any auto eois, right?